### PR TITLE
Add python support for presets API

### PR DESCRIPTION
**Pull request by johningve**
The presets API added in Zivid SDK 2.9 allows the user to obtain settings presets that are fine tuned for each camera model for a number of different use-cases.

See samples/sample_presets.py for an example of how to use it.

### CODE REVIEW

File: modules/zivid/presets.py
Comment (Ayxan13) [inline modules/zivid/presets.py:144]: I don't like using `getattr` here as it relies on names of the fields being the same as their values. How about this:
```suggestion
    return [
        Category(c) for c in _zivid.presets.categories(_to_internal_camera_info(CameraInfo(model=model)).model)
    ]
```

File: modules/zivid/presets.py
Comment (Ayxan13) [inline modules/zivid/presets.py:84]: `type(_zivid.presets.Category)` is similarly wrong here.

File: modules/zivid/presets.py
Comment (Ayxan13) [inline modules/zivid/presets.py:32]: `type(_zivid.presets.Preset)` is wrong here (it's always type). You want just `_zivid.presets.Preset`.

File: modules/zivid/presets.py
Comment (Ayxan13) [reply modules/zivid/presets.py:84]: Also, consider using fstrings instead of string.format

**Review by Ayxan13 — CHANGES_REQUESTED** [review]

File: modules/zivid/presets.py
Comment (eskaur) [reply modules/zivid/presets.py:32]: I would just write "expected zivid.Preset" in the string actually. Something has to be hardcoded anyway, and `_zivid.presets.Preset` inside format() isn't any better.

File: samples/sample_presets.py
Comment (eskaur) [inline samples/sample_presets.py:32]: Love this sample. Maybe also add this at the end:

```
settings_file = chosen_preset.name + ".yml"
print(f"Saving Settings to {settings_file}")
chosen_preset.settings.save(settings_file)
```

File: modules/zivid/presets.py
Comment (eskaur) [inline modules/zivid/presets.py:15]: Do you mean "to obtain a Preset instance"?

File: modules/zivid/presets.py
Comment (eskaur) [reply modules/zivid/presets.py:84]: See the other similar comment for my opinion.

Regarding f-strings: We currently have no f-strings in this package at all. The reason is that we used to support Python 3.5, which doesn't have f-strings. Let's just keep it consistent for now, and then we can upgrade to f-strings later.

File: modules/zivid/presets.py
Comment (eskaur) [inline modules/zivid/presets.py:119]: Stray word "scenarios" after sentence.

File: test/test_presets.py
Comment (eskaur) [inline test/test_presets.py:4]: Maybe just iterate the test over all `zivid.CameraInfo.Model.valid_values()`.

File: modules/zivid/presets.py
Comment (johningve) [reply modules/zivid/presets.py:32]: This __init__ function is almost an exact copy of the init functions in `camera.py` and others. I guess they should all be changed. Maybe in a follow up PR?

Comment (Ayxan13) [reply modules/zivid/presets.py:32]: > Something has to be hardcoded anyway

@eskaur If the class gets renamed later no, your string will be quietly wrong but the format version will complain. I vote for using a runtime formatting, the overhead should be negligible.

Comment (Ayxan13) [reply modules/zivid/presets.py:32]: > I guess they should all be changed. Maybe in a follow up PR?
@johningve Yes, probably. But do fix this particular instance before merging, we shouldn't merge it with the bug :)

**Comment by apartridge** [comment]
We also need to double-check that we are able to add 2D settings presets in the future when we want to without breaking any existing stuff. In the C++ API it is covered since CategoryBase and PresetBase are templated types, and I think it would also work quite well here since Python is very dynamic :)

**Comment by johningve** [comment]
> We also need to double-check that we are able to add 2D settings presets in the future when we want to without breaking any existing stuff. In the C++ API it is covered since CategoryBase and PresetBase are templated types, and I think it would also work quite well here since Python is very dynamic :)

Yeah I don't think there will be any problem. It looks to me like adding 2D settings presets would add new types for 2D presets and categories, and would also require a new function to get the categories. So that could all be added without any conflict to the python wrapper. We will probably be able to reuse code for the preset and category classes.

File: modules/zivid/presets.py
Comment (johningve) [reply modules/zivid/presets.py]: Interestingly, type() does change the output of this error:

With type()

```
TypeError: Unsupported type for argument impl. Got <class 'list'>, expected <class 'pybind11_builtins.pybind11_type'>
```

Without type()
```
TypeError: Unsupported type for argument impl. Got <class 'list'>, expected <class '_zivid._zivid.presets.Category'>
```

I don't know why `_zivid` gets repeated here

Comment (johningve) [reply modules/zivid/presets.py]: > > Something has to be hardcoded anyway
> 
> @eskaur If the class gets renamed later on, your string will be quietly wrong but the format version will complain. I vote for using runtime formatting version, the overhead should be negligible.

:+1:

File: test/test_presets.py
Comment (johningve) [reply test/test_presets.py]: > `assert container` could be a nicer way of doing `assert len(container) > 0`

Ah I see that an empty list evaluates to false. Now that makes sense.

**Comment by johningve** [comment]
So, there are two things to fix in a follow-up PR:

1. Remove the `type(_zivid....)` in the error thrown by initializers for certain classes
2. Scope the camera in the samples

**Comment by johningve** [comment]
Ready for next round

File: samples/sample_presets.py
Comment (eskaur) [reply samples/sample_presets.py]: Then all the other samples are broken as well. I suggest we keep it consistent for now, and then SDK team can follow up with a PR that improves them all in one go.

But @apartridge can decide on this tradeoff with consistency.

File: modules/zivid/presets.py
Comment (eskaur) [reply modules/zivid/presets.py]: Okay I agree :)

**Review by eskaur — APPROVED** [review]
From my perspective this is fine given that the follow-up PRs also get done. Nice work :+1:

**Review by Ayxan13 — APPROVED** [review]
Looks good to me. I would scope the `Application` instance in the sample, but it's not a deal-breaker since we scopped the camera. It would be nice if one of the static linters could warn when objects with `__exit__` method are used without the `with` statement.

File: samples/sample_presets.py
Comment (apartridge) [reply samples/sample_presets.py:36]: @FYI @SatjaSivcev, we are adding a presets sample for Python here, we may want to port this over to https://github.com/zivid/zivid-python-samples.
Can CSU make an issue for this?

**Pull request merged**
